util.go

```diff
@@ -125,6 +125,22 @@ func StrSliceContains(haystack []string, needle string) bool {
 	return false
 }
 
+func StrSliceContainsAll(subset, superset []string) bool {
```

Comment:
For tag filtering logic, I am assuming we need to compare two slices. In our case a `WithTags` filter is the subset, and the existing list of `Tags` is the superset.

Everything passed as `WithTags` needs to be contained within existing `Tags`, otherwise we filter.